### PR TITLE
Fix device selection in PipelinedExecutor.

### PR DESCRIPTION
**Pull request by mzient**
CUDA device id was not properly set in executor (it always tried to run on device 0). It's been fixed with RAII DeviceGuard in all relevant functions.

Signed-off-by: Michal Zientkiewicz <michalz@nvidia.com>

### CODE REVIEW

File: dali/pipeline/executor/executor.cc [hunk 1]
```diff
@@ -75,6 +75,7 @@ void Executor::RunCPU() {
   lock.unlock();
 
   // Run the support ops
+  DeviceGuard g(device_id_);
```
Comment (JanuszL) [inline dali/pipeline/executor/executor.cc]: Please put it above comment, now it is misleading.

Comment (mzient) [reply dali/pipeline/executor/executor.cc]: Fixed.

**Comment by mzient** [comment]
Added `DeviceGuard` to `Executor::Build`. `PipelineExecutor::Build` calls only two public functions, both of which already have `DeviceGuard`, so I didn't put it there.

Build 591988 almost passed (i.e. lint passed and tests passed on most of the targets, some are still queued).

File: dali/pipeline/executor/executor.cc [hunk 2]
```diff
@@ -373,6 +379,7 @@ void Executor::PruneUnusedGraphNodes() {
 void Executor::SetupDataForGraph(WorkspaceBlob *wsb) {
   // Clear any old data setup
   wsb->Clear();
+  DeviceGuard g(device_id_);
```
Comment (Kh4L) [inline dali/pipeline/executor/executor.cc]: Maybe L382 -> L380, for consistency with L275 ?

Comment (mzient) [reply dali/pipeline/executor/executor.cc]: Fixed.

**Pull request merged**